frappe/core/doctype/user/test_user.py

```diff
@@ -371,6 +372,19 @@ def test_user_onload_modules(self):
 		doc = frappe.response.docs[0]
 		self.assertListEqual(doc.get("__onload").get('all_modules', []),
 			[m.get("module_name") for m in get_modules_from_all_apps()])
+	
+	def test_reset_password_link_expiry(self):
+		new_password = "new_password"
+
+		# set the reset password expiry to 1 second
+		frappe.db.set_value("System Settings", "System Settings", "reset_password_link_expiry_seconds", 1)
+		frappe.db.commit() # nosemgrep: this is to set value for a test case
```

Comment:
test should pass without a commit too. If it's in the same transaction future queries will see this modification. Adding commit causes confusing behavior and often introduces flake in tests.  

```suggestion
```

Reply:
@ankush Agreed. Removed the same.